### DISCUSSION TITLE
Rootless installation, CustomDir

### DISCUSSION THREAD

**Discussion by ahmedrowaihi** [coollabsio/coolify#1759] [Feature Requests]
Coolify currently installs, runs docker, sets /data/coolify as in root directory.

# The problem:
Coolify enforces using root user, this might be cursed at some points as it restricts user management, Root takes different context in docker, also other managed .sh docker(compose) scripts all points to /data/coolify which makes it harder to do it rootlessly 

# Solution:
Add optional variable to the workingDir which allows user to choose other directory for installation, and for other docker file, there should be a hook that replaces the /data/coolify with the workingDir specified if exsist

**Comment by roo12312** [comment]
I don't think it's just workingDir that needs to be changed... we need to change the docker.sock location also

**Comment by peaklabs-dev** [comment]
- [ ] Fix none root user permissions
- [ ] do not hardcode root for SSH keys 
- [ ] full binary permission -> not no password like described in the docs https://coolify.io/docs/knowledge-base/server/non-root-user
- [ ] also support rootless docker/ podman
- [ ] also allow Setting a custom install dir

**Comment by unleashit** [comment]
Just ran into this, and for me requiring the root user is a show stopper (as is the workaround). Disappointed because I really wanted to try this. I also don't understand why the choice was made to require SSH even on localhost.

**Comment by fend25** [comment]
100% agree. Even potential root access is a showstopper.

Rootless Docker/Podman is now an industry standard, and there's a need for an easy and robust tool to run it, ensuring it operates entirely in userspace.

**Comment by clemwo** [comment]
Noob here. 
I wanted to disable SSH root access for security reasons. I just tried that which caused my server to become inaccessible. Is that expected?

**Comment by ahmedrowaihi** [comment]
[Dokploy](https://github.com/Dokploy/dokploy)
